tests/baselines/reference/ambientShorthand_reExport.js

```diff
@@ -20,11 +20,19 @@ x($);
 "use strict";
 exports.__esModule = true;
 var jquery_1 = require("jquery");
-exports.x = jquery_1.x;
+Object.defineProperty(exports, "x", { enumerable: true, get: () => jquery_1.x });

```

Comment:
In cases like this it actually should be possible to hoist the `Object.defineProperty` call itself to happen before any of the other `require` calls.

This might make circular imports more compatible.

Reply:
Digging into this, it looks like you are correct - this should be hoisted.  But so should every exported value, whether it's defined as a getter or not.  I'm thinking all these cases should be handled as a separate PR.

As it stands, if the property is accessed but the module from which it was re-exporting the value has not been loaded, we won't get the value in either case.  The only thing that's affected is enumerability, which is already an issue without this PR.  This PR is a step in the right direction and solves #12522.  The hoisting is a separate issue IMO.